File: backoffice/src/shared/domain/AggregateRoot.py
"""
 *
 * Libraries 
 *
"""

from .events.DomainEvent import DomainEvent

"""
 *
 * Class 
 *
"""

class AggregateRoot:

    """
     *
     * Attributes 
     *
    """

    __domainEvents : list[ DomainEvent ]

    """
     *
     * Methods 
     *
    """

    def __init__( self ) -> object:
        self.__domainEvents = []
    
    def recordEvent( self, domainEvent : DomainEvent ) -> None:
        self.__domainEvents.append( domainEvent )
    
    def pullEvents( self ) -> list[ DomainEvent ]:
        # Variables
        events : list[ DomainEvent ]
        # Code
        events = self.__domainEvents
        self.__domainEvents = []
        return events

File: preparation/src/shared/domain/AggregateRoot.cs
/* 
 * 
 * Class
 *
*/

namespace preparation.src.shared.domain {

    public class AggregateRoot {

        /* 
         * 
         * Attributes
         *
        */

        private List<DomainEvent> domainEvents;

        /* 
         * 
         * Methods
         *
        */

        public AggregateRoot() {
            domainEvents = new List<DomainEvent>();
        }

        public void recordEvent( DomainEvent domainEvent ) {
            domainEvents.Add( domainEvent );
        }

        public List<DomainEvent> pullEvents() {
            // Variables
            List<DomainEvent> events;
            // Code
            events  = domainEvents;
            domainEvents = new List<DomainEvent>();
            return events;
        }

    }

}